src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/ManifestRecord.cs

```diff
@@ -9,20 +9,39 @@ namespace Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
 
 public abstract class ManifestRecord : IComparable<ManifestRecord>
 {
-    public string Path { get; set; } = null!;
-    public string RemoteUri { get; set; } = null!;
-    public string CommitSha { get; set; } = null!;
+    public string Path { get; set; }
+    public string RemoteUri { get; set; }
+    public string CommitSha { get; set; }
+
+    public ManifestRecord(string path, string remoteUri, string commitSha)
+    {
+        Path = path;
+        RemoteUri = remoteUri;
+        CommitSha = commitSha;
+    }
 
     public int CompareTo(ManifestRecord? other)
     {
-        if (other == null) return 1;
+        if(other == null) return 1;
```

Comment:
nit: I think the space is what we usually do.

But also I think it's preferred to use braces everywhere:

```suggestion
        if (other == null)
        {
            return 1;
        }

```